File: cc/stack/MaximumNestingDepthOfTheParentheses.cc
#include <string>
#include <deque>

using namespace std;

/**
 * LeetCode 1614. Maximum Nesting Depth of the Parentheses
 * https://leetcode.com/problems/maximum-nesting-depth-of-the-parentheses/
 */
class MaximumNestingDepthOfTheParentheses {
public:
    int maxDepth(string s) {
        int currentDepth = 0;
        int maximumDepth = 0;
        deque<char> aStack;
        for (int i = 0; i < s.size(); i++) {
            switch (s[i]) {
            case '(':
                aStack.push_back('(');
                currentDepth++;
                break;
            case ')':
                aStack.pop_back();
                currentDepth--;
                break;
            default:
                break;
            }
            maximumDepth = max(maximumDepth, currentDepth);
        }
        return maximumDepth;
    }
};

int main(int argc, char const *argv[]) {
    // ...
    return 0;
}

/* EOF */

File: java/src/io/zihengcat/leetcode/stack/MaximumNestingDepthOfTheParentheses.java
package io.zihengcat.leetcode.stack;

import java.util.Deque;
import java.util.LinkedList;

/**
 * LeetCode 1614. Maximum Nesting Depth of the Parentheses
 * https://leetcode.com/problems/maximum-nesting-depth-of-the-parentheses/
 */
public class MaximumNestingDepthOfTheParentheses {
    public static void main(String[] args) {
        // ...
    }
    public int maxDepth(String s) {
        if (s == null) {
            return 0;
        }
        // Deque<Character> aStack = new LinkedList<>();
        int maximumDepth = 0;
        int currentDepth = 0;
        for (int i = 0; i < s.length(); i++) {
            switch (s.charAt(i)) {
            case '(':
                // aStack.push('(');
                currentDepth++;
                break;
            case ')':
                // aStack.pop();
                currentDepth--;
            default:
                break;
            }
            maximumDepth = Math.max(maximumDepth, currentDepth);
        }
        return maximumDepth;
    }
}

/* EOF */
